### ISSUE TITLE
Installation fails due to conflicting numpy version

### ISSUE THREAD

**Issue by NeolithEra** [ViaFerrata/OrcaSong#1]
Hi, users are unable to run OrcaSong due to dependency conflict with _**numpy**_ package. 

As shown in the following full dependency graph of OrcaSong, OrcaSong requires _**numpy**_ ，while **_km3pipe 8.19.1_** requires _**numpy==1.16.2**_.

According to pip’s “first found wins” installation strategy, _**numpy 1.16.4**_ is the actually installed version. However, _**numpy 1.16.4**_ does not satisfy _**numpy==1.16.2**_.

### Dependency tree------
```
orcasong<version range:>
| +-docopt<version range:>
| +-h5py<version range:>
| +-km3pipe<version range:>
| | +-docopt<version range:>
| | +-h5py<version range:>
| | +-ipython<version range:>
| | +-ipywidgets<version range:>
| | +-matplotlib<version range:>=2.2.0>
| | +-mock<version range:>
| | +-numpy<version range:==1.16.2>
| | +-numpydoc<version range:>=0.7.0>
| | | +-jinja2<version range:>=2.3>
| | | +-sphinx<version range:>=1.2.3>
| | +-pandas<version range:>
| | +-pillow<version range:>
| | +-pydocstyle<version range:>
| | +-pytest<version range:>
| | +-pytest-cov<version range:>
| | +-pytest-flake8<version range:>
| | +-pytest-pylint<version range:>
| | +-pytest-watch<version range:>
| | +-requests<version range:>
| | +-scipy<version range:==1.2.1>
| | +-seaborn<version range:>
| | +-setuptools<version range:>=40.6.2>
| | +-setuptools-scm<version range:>
| | +-sklearn<version range:>
| | | +-scikit-learn<version range:>
| | +-sphinx<version range:>=1.6.3>
| | +-sphinx-autoapi<version range:>
| | +-sphinx-gallery<version range:>=0.1.12>
| | | +-matplotlib<version range:>
| | | +-pillow<version range:>
| | | +-setuptools<version range:>
| | | +-sphinx<version range:>
| | +-sphinx-rtd-theme<version range:>=0.3>
| | +-sphinxcontrib-napoleon<version range:>=0.6.1>
| | +-sphinxcontrib-programoutput<version range:>=0.11>
| | +-sphinxcontrib-websupport<version range:>=1.0.1>
| | +-statsmodels<version range:==0.9.0>
| | +-tables<version range:>=3.4.2>
| | +-toml<version range:>
| | +-tornado<version range:>
| | +-tqdm<version range:>
| | +-twine<version range:>
| | +-urwid<version range:>
| | +-websocket-client<version range:>
| | | +-six<version range:>
| | +-yapf<version range:>=0.26>
| +-matplotlib<version range:>
| +-numpy<version range:>
| +-numpydoc<version range:>
| | +-jinja2<version range:>=2.3>
| | +-sphinx<version range:>=1.2.3>
| +-setuptools-scm<version range:>
| +-sphinx<version range:>
| +-sphinx-autoapi<version range:>
| +-sphinx-rtd-theme<version range:>
| +-toml<version range:>
| +-twine<version range:>
```

Thanks for your help.
Best,
Neolith

**Comment by NeolithEra** [comment]
### Solution

1.	Fix your direct dependency to be numpy==1.16.2 I have checked this revision will not affect your downstream projects now.
2.	Ask your upstream project km3pipe to loose the version range of numpy.
Which solution do you prefer, 1 or 2?

@ViaFerrata Please let me know your choice. I can submit a PR to solve this issue.

**Comment by tamasgal** [comment]
Btw. just for the sake of completeness, the freeze of `numpy==1.16.2` is due to a serious memory leak which appeared in `numpy v1.16.3+` and is planned to be fixed in `v1.17`: https://github.com/numpy/numpy/issues/13853

**Comment by ViaFerrata** [comment]
@NeolithEra 
Thanks for the notice. Would be appreciated, if you just do 1. (include it in OrcaSong's requirements) and submit a PR.
